portal/flatpak-portal.c

```diff
@@ -43,6 +47,41 @@ static GMainLoop *main_loop;
 static PortalFlatpak *portal;
 static gboolean opt_verbose;
 
+G_LOCK_DEFINE (update_monitors);
+static GHashTable *update_monitors;
+static guint update_monitors_timeout = 0;
+static gboolean update_monitors_timeout_running_thread = FALSE;
+
+/* Poll all update monitors twice an hour */
+#define UPDATE_POLL_TIMEOUT_SEC (30 * 60)
```

Comment:
Typically we only pull the summary.sig and if that didn't change we don't pull the entire summary, so this will only happen when it changes. Of course, some repos *do* change often.

But anyway, something else (like gnome-software) will do a similar poll on the side so we'll be downloading it anyway, meaning the cache would then be hit.

I guess we might want some kind of configuration option in the installation for a default poll rate to use, that way gnome-software and the portal would use the same delay.

Reply:
I guess in the p2p non-signed-summary case we might be downloading the entire summary each time, or do we use an ETAG for that?

Reply:
Looks like `OstreeRepoFinderMount` iterates over the `refs/` subdirectories. I don't think `OstreeRepoFinderAvahi` uses ETag but anyway I'm not concerned about data transfer over a local network or to a USB drive since those tend to be faster than Internet transfer and not metered.

Reply:
> How does the update polling implementation here fit in with gnome-software’s automatic polling for updates? What if they both start installing an update at the same time — is the flatpak dir locked? 

We already have to be safe for parallel updates, since the user can just run flatpak update on the commandline at any time. So yes, that should be safe.

> More generally, what are the use cases for this portal? What problems does it solve? Allowing apps to explicitly check for updates from inside the app (like a Help → Check for Updates menu entry); or constantly checking for updates in the background and allowing the app to know when they are installed so it can prompt the user to restart the app?

More the former. You can already monitor for updates being installed, using the /app/.updated file.